### PR TITLE
Sign

### PR DESCRIPTION
**Pull request by alewis**
Adds sign and abs backend functions

### CODE REVIEW

File: tensornetwork/backends/abstract_backend.py [hunk 1]
```diff
@@ -734,3 +734,29 @@ def matmul(self, tensor1: Tensor, tensor2: Tensor) -> Tensor:
...
+    """
+    raise NotImplementedError(
+        "Backend '{}' has not implemented `abs`.".format(self.name))
+
+  def sign(self, tensor: Tensor):
+    """
+    Returns an elementwise tensor with entries
+    y[i] = 1, 0, -1 tensor[i] > 0, == 0, and < 0 respectively.
+
+    For complex input the behaviour of this function may depend on the backend.
```
Comment (Thenerdstation) [inline tensornetwork/backends/abstract_backend.py:754]: Don't include details of other backends in the abstract backend.

File: tensornetwork/backends/jax/jax_backend.py [hunk 2]
```diff
@@ -674,3 +674,29 @@ def matmul(self, tensor1: Tensor, tensor2: Tensor) -> Tensor:
...
+
+    For complex input the behaviour of this function may depend on the backend.
+    With NumPy and Jax, it returns y[i] = x[i]/sqrt(x[i]^2). In
+    TensorFlow it returns y[i] = x[i] / abs(x[i]). In PyTorch it is
+    not implemented.
+
+    Args:
+      tensor: The input tensor.
+    """
+    out = jnp.sign(tensor)
```
Comment (Thenerdstation) [inline tensornetwork/backends/jax/jax_backend.py:701]: `return jnp.sign(tensor)`

File: tensornetwork/backends/numpy/numpy_backend.py [hunk 3]
```diff
@@ -621,3 +621,29 @@ def matmul(self, tensor1: Tensor, tensor2: Tensor) -> Tensor:
...
+    For complex input the behaviour of this function may depend on the backend.
+    With NumPy and Jax, it returns y[i] = x[i]/sqrt(x[i]^2). In
+    TensorFlow it returns y[i] = x[i] / abs(x[i]). In PyTorch it is
+    not implemented.
+
+    Args:
+      tensor: The input tensor.
+    """
+    out = np.sign(tensor)
+    return out
```
Comment (Thenerdstation) [inline tensornetwork/backends/numpy/numpy_backend.py:649]: `return np.sign(tensor)`

File: tensornetwork/backends/tensorflow/tensorflow_backend.py [hunk 4]
```diff
@@ -267,3 +267,29 @@ def matmul(self, tensor1: Tensor, tensor2: Tensor) -> Tensor:
...
+    For complex input the behaviour of this function may depend on the backend.
+    With NumPy and Jax, it returns y[i] = x[i]/sqrt(x[i]^2). In
+    TensorFlow it returns y[i] = x[i] / abs(x[i]). In PyTorch it is
+    not implemented.
+
+    Args:
+      tensor: The input tensor.
+    """
+    out = tf.math.sign(tensor)
+    return out
```
Comment (Thenerdstation) [inline tensornetwork/backends/tensorflow/tensorflow_backend.py:295]: `return tf.math.sign(tensor)`

**Comment by alewis** [comment]
Comments should be addressed

File: tensornetwork/backends/abstract_backend.py [hunk 5]
```diff
@@ -734,3 +734,25 @@ def matmul(self, tensor1: Tensor, tensor2: Tensor) -> Tensor:
...
+    Returns:
+      tensor: Its elementwise absolute value.
+    """
+    raise NotImplementedError(
+        "Backend '{}' has not implemented `abs`.".format(self.name))
+
+  def sign(self, tensor: Tensor):
+    """
+    Returns an elementwise tensor with entries
+    y[i] = 1, 0, -1 tensor[i] > 0, == 0, and < 0 respectively.
```
Comment (mganahl) [inline tensornetwork/backends/abstract_backend.py:752]: missing "for" or "if"

File: tensornetwork/backends/jax/jax_backend.py [hunk 6]
```diff
@@ -674,3 +674,26 @@ def matmul(self, tensor1: Tensor, tensor2: Tensor) -> Tensor:
...
+      tensor: An input tensor.
+    Returns:
+      tensor: Its elementwise absolute value.
+    """
+    return jnp.abs(tensor)
+
+  def sign(self, tensor: Tensor) -> Tensor:
+    """
+    Returns an elementwise tensor with entries
+    y[i] = 1, 0, -1 tensor[i] > 0, == 0, and < 0 respectively.
```
Comment (mganahl) [inline tensornetwork/backends/jax/jax_backend.py:691]: missing "for" or "if"

File: tensornetwork/backends/jax/jax_backend_test.py [hunk 7]
```diff
@@ -787,3 +787,23 @@ def A_mv(x):
   rtol = tol*jax.numpy.linalg.norm(b)
   atol = tol
   assert err < max(rtol, atol)
+
+
+@pytest.mark.parametrize("dtype", np_dtypes)
+def test_abs(dtype):
+  shape = (4, 3, 2)
```
Comment (mganahl) [inline tensornetwork/backends/jax/jax_backend_test.py:794]: set seed

Comment (mganahl) [reply tensornetwork/backends/jax/jax_backend_test.py:794]: everywhere

File: tensornetwork/backends/numpy/numpy_backend_test.py [hunk 8]
```diff
@@ -847,3 +847,23 @@ def test_matmul():
   actual = backend.matmul(a, b)
   expected = np.matmul(t1, t2)
   np.testing.assert_allclose(expected, actual)
+
+
+@pytest.mark.parametrize("dtype", np_dtypes)
```
Comment (mganahl) [inline tensornetwork/backends/numpy/numpy_backend_test.py:852]: seeds

**Comment by mganahl** [comment]
can you fix the conflicts? Sry I may have screwed this one up. Was one of the other PRs depending on this one?

**Comment by alewis** [comment]
All my recent PRs regarding backends edit basically the same file because they were split from a bigger commit. My Git dexterity wasn't sufficient to avoid this.

**Comment by mganahl** [comment]
OK np

File: tensornetwork/backends/tensorflow/tensorflow_backend_test.py [hunk 9]
```diff
@@ -515,6 +515,26 @@ def test_matmul():
   np.testing.assert_allclose(expected, actual)
 
 
+@pytest.mark.parametrize("dtype", tf_dtypes)
+def test_abs(dtype):
+  shape = (4, 3, 2)
```
Comment (mganahl) [inline tensornetwork/backends/tensorflow/tensorflow_backend_test.py:520]: seeds, can you check every test and add seeds to all of them? thanks

Comment (alewis) [reply tensornetwork/backends/tensorflow/tensorflow_backend_test.py:520]: Ok, I think (hope?) I got them all
